Microsoft.Toolkit.Uwp.UI.Controls/WrapPanel/WrapPanel.cs

```diff
@@ -36,18 +79,34 @@ public Orientation Orientation
         /// </summary>
         public static readonly DependencyProperty OrientationProperty =
             DependencyProperty.Register(
-                "Orientation",
+                nameof(Orientation),
                 typeof(Orientation),
                 typeof(WrapPanel),
-                new PropertyMetadata(Orientation.Horizontal, OrientationPropertyChanged));
+                new PropertyMetadata(Orientation.Horizontal, LayoutPropertyChanged));
 
-        private static void OrientationPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        private static void LayoutPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
         {
-            var wrapPanel = d as WrapPanel;
-            wrapPanel?.InvalidateMeasure();
-            wrapPanel?.InvalidateArrange();
+            if (d is WrapPanel wp)
+            {
+                if (wp.Orientation == Orientation.Horizontal)
+                {
+                    wp._uSpacing = wp.HorizontalSpacing;
+                    wp._vSpacing = wp.VerticalSpacing;
+                }
+                else
+                {
+                    wp._uSpacing = wp.VerticalSpacing;
+                    wp._vSpacing = wp.HorizontalSpacing;
+                }
+
+                wp.InvalidateMeasure();
+                wp.InvalidateArrange();
+            }
         }
 
+        private double _uSpacing = 0d;
```

Comment:
Add these to the UvMeasure class and remove the if above

Reply:
Haven't done this exactly, but have changed to use UVMeasure for spacing and removed the if and fields.